src/dosbox.cpp

```diff
@@ -1112,6 +1112,21 @@ void DOSBOX_Init()
 	pstring->Set_help(
 	        "Apply Y-axis calibration parameters from the hotkeys ('auto' by default).");
 
+	pbool = secprop->Add_bool("hotplug", when_idle, true);
+	pbool->Set_help(
+	        "Automatically detect when a game controller is connected or disconnected\n"
+	        "(enabled by default).\n"
+	        "Currently, when this option is enabled, bindings will be reinitialized from the\n"
+	        "configured mapper file (or from defaults if it does not exist), so any unsaved\n"
+	        "binding changes made in the mapper UI will be lost whenever a controller is\n"
+	        "connected or disconnected. So you may wish to disable this option at least\n"
+	        "temporarily if you have issues losing unsaved bindings before you can save them,\n"
+	        "e.g. due to a misbehaving game controller that frequently connects /\n"
+	        "disconnects.\n"
+	        "When this option is disabled, only game controllers connected at startup will be\n"
+	        "visible to DOSBox, and any controllers that become disconnected after startup\n"
+	        "will not be available until the application is restarted.\n");
```

Comment:
Nice help text 💯, but we prefer to introduce as few extra manual options as possible. I'd say the existence of this setting and its description text "smells" of a workaround, hence my strong preference is to always save as explained in the comment.

Just to save you from potentially doing further throwaway work, it's OK if you write any such setting descriptions as the very last step, once we've agreed on the final solution.

Reply:
Makes sense. For now I left settings-related changes in https://github.com/arifogel/dosbox-staging/tree/ari-hotplug-joystick-with-setting and removed them from the PR. Easy to put back if needed.